Apromore-Extras/APMLogModule/src/main/java/org/apromore/apmlog/filter/PTrace.java

```diff
@@ -415,7 +417,14 @@ private AActivity getActivity(int index, int fromIndex, IntArrayList markedIndex
 
                             break;
                         } else {
-                            eventIndexList.add(i);
+                            if (lifecycle.equals("start")){
+                                if (!foundStart) {
+                                    eventIndexList.add(i);
+                                    foundStart = true;
+                                }
+                            } else {
+                                eventIndexList.add(i);
+                            }
                         }
                     }
                 }
```

Comment:
This code is duplicated with the code above in LogFactory.java (getActivity). Can we do a bit refactoring here to remove this duplication?